torchdata/dataloader2/dataloader2.py

```diff
@@ -49,7 +49,11 @@ def __next__(self) -> T_co:
         if self.iterator_id == self.dataloader.valid_iterator_id:
             self.dataloader._reset_iter = True
             try:
-                if self.dataloader._is_paused:
+                if self.dataloader._is_paused or (
+                    self.limit_threshold is not None and self.limit_counter >= self.limit_threshold
+                ):
+                    if not self.dataloader._is_paused:
+                        self._pause()
```

Comment:
This is an optional change to make `pause` lazily at the time of `limit+` is reached.